MainApp/Servers/vehicle/src/main/java/vehicle/service/impl/ModelServiceImpl.java

```diff
@@ -1,41 +1,130 @@
 package vehicle.service.impl;
 
+import org.dozer.DozerBeanMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import vehicle.dto.ModelDTO;
+import vehicle.exceptions.ConversionFailedError;
+import vehicle.exceptions.DuplicateEntity;
+import vehicle.exceptions.EntityNotFound;
+import vehicle.model.Brand;
+import vehicle.model.Model;
+import vehicle.repository.BrandRepo;
 import vehicle.repository.ModelRepo;
 import vehicle.service.ModelService;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class ModelServiceImpl implements ModelService {
 
     @Autowired
     ModelRepo modelRepo;
 
+    @Autowired
+    BrandRepo brandRepo;
+
+    @Autowired
+    DozerBeanMapper mapper;
+
+    public ModelDTO convertToDTO(Model model) throws ConversionFailedError {
+        try {
+            return mapper.map(model, ModelDTO.class);
+        } catch (Exception e) {
+            throw new ConversionFailedError("Internal server error");
+        }
+    }
+
+    @Override
+    public Model convertToModel(ModelDTO modelDTO) throws ConversionFailedError {
+        try {
+            return mapper.map(modelDTO, Model.class);
+        } catch (Exception e) {
+            throw new ConversionFailedError("Invalid data");
+        }
+    }
+
     @Override
-    public List<ModelDTO> getAll(Long brandId) {
-        return null;
+    public List<ModelDTO> getAll(Long brandId) throws EntityNotFound, ConversionFailedError {
```

Comment:
Ovde treba samo modele od brenda da vrati, msm tako je u kontroleru.

Reply:
ou šit